### PR TITLE
Update PR checklist

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR updates the PR checklist to include the check for updating the Kibana version constraint.

### CODE REVIEW

**Comment by mtojek** [comment]
Yes, that's the goal, Christos. The hard part is how to phrase it and where, to make sure that people will look into this :) The version constraint should be updated if you aren't introducing any bugfixes to the current one, but adding new features. Otherwise we would have to run compatibility tests against older stack version (I believe we wouldn't like to do this).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Explain here the changes you made on the PR.
 - [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
 - [ ] I have verified that all data streams collect metrics or logs.
 - [ ] I have added an entry to my package's `changelog.yml` file.
+- [ ] As I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).
```
Comment (jsoriano) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: Should this say `If...` instead of `As...`?

Comment (jsoriano) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: Out of curiosity, are there other possible conditions? I see we only use `kibana.version` so far.

**Review by jsoriano — APPROVED** [review]
Looks good, only a couple of questions.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (mtojek) [reply .github/PULL_REQUEST_TEMPLATE.md:20]: There are plans for same constraint for the Elastic-Agent binary, but it's not supported now by all parties.

Comment (mtojek) [reply .github/PULL_REQUEST_TEMPLATE.md]: Right, fixed!

**Comment by andresrc** [comment]
✅ from me as well. Even though our goal is to be able to make releases out of band, for this initial iterations we should focus on in-band releases and use out-of-band only for bug-fixes. We need to clarify the guidelines here, which will evolve as the solution matures, of course.

**Pull request merged**